code/src/components/Form.js

```diff
@@ -0,0 +1,305 @@
+import React, { useState } from 'react';
+
+import InputText from './InputText';
+import InputSelect from './InputSelect';
+import InputCheckbox from './InputCheckbox';
+import InputRadiobutton from './InputRadiobutton';
+import Summary from './Summary';
+import Button from './Button';
+import ProgressBar from './ProgressBar';
+import '../styles/Form.scss';
+
+const Form = () => {
+  const [pageIndex, setPageIndex] = useState(-1);
+  const [text, setText] = useState('');
+  const [select, setSelect] = useState('');
+  const [radiobutton, setRadio] = useState();
+  const [checkbox, setCheckbox] = useState([]);
+  const [submit, setSubmit] = useState(false);
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    setSubmit(true);
+  };
+  const handleTextChange = event => {
+    setText(event.target.value);
+  };
+  const handleSelectChange = event => {
+    setSelect(event.target.value);
+  };
+  const handleRadioChange = event => {
+    setRadio(event.target.value);
+  };
+  const handleCheckboxChange = checkboxValue => {
+    checkbox.includes(checkboxValue)
+      ? setCheckbox(checkbox.filter(item => item !== checkboxValue))
+      : setCheckbox([...checkbox, checkboxValue]);
+  };
+
+  const onNextPageChange = () => {
+    setPageIndex(pageIndex + 1);
+  };
+  const onPreviousPageChange = () => {
+    setPageIndex(pageIndex - 1);
+  };
+  const onRefresh = () => {
+    window.location.reload();
+  };
+
+  const selectArray = [
```

Comment:
I'm very happy to see such clean approach with arrays. There is one thing we could improve even more - create separate file, something like `utils.js` or `data.js` and store all arrays and objects there